### PR TITLE
L2 normalize op

### PR DESCRIPTION
**Pull request by shkarupa-alex**
L2 normalization with backend-specific ops for TF and Torch.
There is a universal `nomalize` op, but:
- there is no option to specify epsilon
- this implementation uses backend-supported ops and should have better performance/op fusion compatibility

### CODE REVIEW

**Comment by fchollet** [comment]
We could add the `epsilon` arg in `normalize()`, that would be more general.

> this implementation uses backend-supported ops and should have better performance/op fusion compatibility

If you're worried about that you could just special-case n=2 in `normalize` for the TF backend. It seems this only applies to TF.

**Comment by shkarupa-alex** [comment]
> We could add the `epsilon` arg in `normalize()`, that would be more general.
> If you're worried about that you could just special-case n=2 in `normalize` for the TF backend. It seems this only applies to TF.

Done

**Review by fchollet — COMMENTED** [review]
Thanks for the update -- the changes look good! Please remove `l2_normalize` since it is now strictly equivalent to `normalize`.

**Comment by shkarupa-alex** [comment]
Done, but TF tests failed due to network error

File: keras/src/ops/nn.py [hunk 1]
```diff
@@ -2018,19 +2023,31 @@ def normalize(x, axis=-1, order=2):
...
-    epsilon = backend.epsilon()
+    if epsilon is None:
+        epsilon = backend.epsilon()
+    if 2 == order:
+        # A special case: L2 normalization with `x * rsqrt(...)`
+        # instead of `x / sqrt(...)`
+        square_sum = backend.numpy.sum(
+            backend.numpy.square(x), axis=axis, keepdims=True
+        )
+        inv_norm = backend.math.rsqrt(square_sum)
```
Comment (fchollet) [inline keras/src/ops/nn.py:2048]: Shouldn't the `maximum` op be here, as in `rsqrt(maximum(square_sum, 1e-12))`?

**Review by fchollet — APPROVED** [review]
LGTM, thanks

**Pull request merged**